lib/matplotlib/cbook/__init__.py

```diff
@@ -2194,6 +2194,44 @@ def _check_in_list(_values, **kwargs):
                 .format(v, k, ', '.join(map(repr, values))))
 
 
+def _check_shape(_shape, **kwargs):
+    """
+    For each *key, value* pair in *kwargs*, check that *value* has the shape
+    *_shape*, if not, raise an appropriate ValueError.
+
+    None in the shape is treated as a "free" sizes that can have any length.
+    e.g. (None, 2) -> (N, 2)
+
+    The values checked must be numpy arrays.
+
+    Examples
+    --------
+    To check from (N, 2) shaped arrays
+
+    >>> cbook._check_in_list((None, 2), arg=arg, other_arg=other_arg)
+    """
+    target_shape = _shape
+    for k, v in kwargs.items():
+        data_shape = v.shape
+
+        if len(target_shape) != len(data_shape) or any(
+                t not in [s, None]
+                for t, s in zip(target_shape, data_shape)
+        ):
+            dim_labels = iter(itertools.chain(
+                'MNLIJKLH',
+                (f"D{i}" for i in itertools.count())))
+            text_shape = ", ".join(str(_) for _ in
+                                   (n if n is not None else next(dim_labels)
+                                    for n in target_shape))
```

Comment:
I don't understand why you have two `for`s; isn't this just:
```suggestion
            text_shape = ", ".join(str(n) if n is not None
                                   else next(dim_labels)
                                   for n in target_shape)
```

Reply:
Yup, I was making it more complicated than it needed to be.